### PR TITLE
Making DateTime editable when flag is set

### PR DESCRIPTION
**Pull request by Waize**
We needed the DateTime to be enabled or disabled depending on a boolean flag. 
So I added it to the component. 

As default value it is always enabled to use so it will not change existing uses.

#### What testing has been done on this PR?
In our own application.

### CODE REVIEW

**Comment by alansouzati** [comment]
Thanks for your contribution but I believe this is already supported today with transfer props.

See this line here: https://github.com/grommet/grommet/pull/1034/files#diff-ff44d471ee39c240dc39018a6307606bR288

if you send `<DateTime disabled/>` I'm expecting the input to have that property propagated. 

On a side note, I would be careful with disabling properties. As a user, I often don't know what I have to do in order to enable that field again. We did a bunch of user testings in this area, and our recommendation is not to use disabled fields.

There is an interesting thread on User Experience exchange around a similar topic for disabling Buttons in a form: http://ux.stackexchange.com/questions/9788/disabled-submit-button-on-form-vs-allow-submit-then-show-errors

**Comment by Waize** [comment]
Hi, 

I've already tried the transfer props and they will not work. 
Here is a Codepen:
http://codepen.io/anon/pen/vyOebz?editors=0011

I understand your advice for the disabled fields. 
In our case we do not impelement an public application. The user of our application will be trained to understand why all fields are default disabled. And when they want to edit them they have to enter the "edit"-mode. 
In our (maybe special) case this avoids unwanted data manipulation. 
Thats why we need this disable-Flag on nearly all components. 
And as I said our user know what they have to do if the fields are disabled. 

But thanks for this interesting side note. :)

File: src/js/components/DateTime.js [hunk 1]
```diff
@@ -287,10 +287,11 @@ export default class DateTime extends Component {
       <div ref={(ref) => this.containerRef = ref} className={classes}>
         <input ref={(ref) => this.inputRef = ref} {...props}
```
Comment (alansouzati) [inline src/js/components/DateTime.js]: `{...props}` should have support for disabled

File: src/js/components/DateTime.js [hunk 2]
```diff
@@ -303,7 +304,8 @@ DateTime.contextTypes = {
 
 DateTime.defaultProps = {
   format: 'M/D/YYYY h:mm a',
-  step: 1
+  step: 1,
+  disabled: false
```
Comment (alansouzati) [inline src/js/components/DateTime.js]: I don't think we need to declare a proptype for disabled.

**Comment by alansouzati** [comment]
Cool, let me try and investigate why the codepen is not working.

**Comment by alansouzati** [comment]
You codepen does disable the input text field. I'm closing this pull request as the code in place already does the job that this pr is proposing.

Custom Grommet components will not support disabled properties directly, it is not part of our pattern, I'm sorry.

screen shot 2016-11-09 at 9 52 21 am

**Comment by Waize** [comment]
Yes you are right, the input field is disabled but not the button. 
So currently you can avoid that the user can enter values but the button still have function. 
That's bad usablity in my opinion. 

So is it possible to make the button disabled too?

**Comment by alansouzati** [comment]
Well, we are technically not disabling anything, but just transferring properties. 

Grommet will not add support explicitly for disabled custom fields. I'm sorry but it is just not part of our UX patterns. In fact, yesterday I was using a form from a big company trying to buy a product. They had the buy button disabled and I entered all the data in the form and I did not realize I entered a wrong data in the form. The button kept disabled and I had no idea what I had to do in order to enable the button again. That was super annoying. After reviewing the form I finally realized that I entered the wrong expiration date. 

Grommet pattern is to ALWAYS enable the form fields and perform post form submission validation.

Can you share what you are trying to achieve with a disabled datetime form field? 

Fortunately, or unfortunately, Grommet is a User Experience framework and there are things we have to be opinionated about, and this is the case here. Hopefully it makes sense for you.

**Comment by Waize** [comment]
Hey Alan, 

I'm understanding your case and I'm sorry to hear that there are people out there who will not give any information about wrong data in forms. 

And I hope you will understand why we want to disable fields. 
In our application we provide forms for managing attributes of documents. In the default-case the user will just see all attributes (see image below). In this view all fields are disabled so the user can not edit the attributes.
doc_view
In this image my modified DateTime-Component was used. 
Now: with your DateTime-Component the user cannot write any data in the date field but still can click the drop-down-button and the calendar opens. This behavior will suggest the user that he can enter some data but clicking on any date will not write the date in the input-field as you can easily see in the codepen. 
And this is why I created this pull request because it's bad usablity in my opionion. I really want to have that the drop-down-button will not open the calendor if it is clicked and disabled.

btw: 
clicking on the edit-button will turn the formfield green and the user can edit the fields (see below).
doc_edit

**Comment by Waize** [comment]
@alansouzati any further points about this? :)

**Pull request closed without merging**